src/NHibernate.Test/SystemTransactions/TransactionFixture.cs

```diff
@@ -29,11 +25,84 @@ public void CanUseSystemTransactionsToCommit()
 			using (ISession session = Sfi.OpenSession())
 			using (TransactionScope tx = new TransactionScope())
 			{
-				W w = session.Get<W>(identifier);
+				var w = session.Get<Person>(identifier);
 				Assert.IsNotNull(w);
 				session.Delete(w);
 				tx.Complete();
 			}
 		}
+
+		[Test]
+		public void FlushFromTransactionAppliesToDisposedSharingSession()
+		{
+			using (var s = OpenSession())
+			{
+				var builder = s.SessionWithOptions().Connection();
+
+				using (var t = new TransactionScope())
```

Comment:
> Never create more than one concurrent ISession or ITransaction instance per database connection.

A bit contradictory with the connection sharing option indeed. By the way, previously was doable only by supplying a user connection I guess.

About having an entity graph from different sessions, yes that does not look as a good practice, I will change that.